osbuild/util/mnt.py

```diff
@@ -5,39 +5,47 @@
 import subprocess
 
 
-def mount(source, target, bind=True, ro=True, private=True, mode="0755"):
+def _run_mount(source, target, args):
+    try:
+        subprocess.run(["mount"] + args + [source, target],
+                       stderr=subprocess.STDOUT,
+                       stdout=subprocess.PIPE,
+                       encoding="utf-8",
+                       check=True)
+    except subprocess.CalledProcessError as e:
+        code = e.returncode
+        msg = e.stdout.strip()
+        raise RuntimeError(f"{msg} (code: {code})") from e
+
+
+def mount(source, target, rbind=True, ro=True, rprivate=True, mode="0755"):
     options = []
     if ro:
         options += ["ro"]
     if mode:
         options += [mode]
 
     args = []
-    if bind:
+    if rbind:
         args += ["--rbind"]
-    if private:
+    if rprivate:
         args += ["--make-rprivate"]
     if options:
         args += ["-o", ",".join(options)]
+    _run_mount(source, target, args)
 
-    r = subprocess.run(["mount"] + args + [source, target],
-                       stderr=subprocess.STDOUT,
-                       stdout=subprocess.PIPE,
-                       encoding="utf-8",
-                       check=False)
-
-    if r.returncode != 0:
-        code = r.returncode
-        msg = r.stdout.strip()
-        raise RuntimeError(f"{msg} (code: {code})")
 
-
-def umount(target, lazy=False):
+def umount(target, lazy=False, recursive=True):
     args = []
+    if recursive:
+        args += ["-R"]
     if lazy:
         args += ["--lazy"]
+    # The sync should in theory not be needed but in rare
```

Comment:
FYI this is what Lorax has ended up with, after hitting similar issues: https://github.com/weldr/lorax/blob/master/src/pylorax/imgutils.py#L262
It also includes the output of fuser to help make it easier to debug things since it's usually nearly impossible to reliably reproduce.